pkgs/development/compilers/haxe/haxe-packages.nix

```diff
@@ -0,0 +1,274 @@
+{ stdenv, fetchzip }:
+
+let
+  withCommas = stdenv.lib.replaceChars ["."] [","];
+in {
+  hxcpp = stdenv.mkDerivation rec {
+    libname = "hxcpp";
+    version = "3.4.64";
+    name = "${libname}-${version}";
+
+    src = fetchzip {
+      inherit name;
+      url = "http://lib.haxe.org/files/3.0/${withCommas name}.zip";
```

Comment:
That 3.0 is haxelib's own api version. That was introduced to separate old libraries that work with Haxe 2.x from the ones that work with Haxe 3.x.

I don't have opinion on making it a variable or not.